### PR TITLE
Use mock network tables in tests

### PR DESCRIPTION
**Pull request by SamCarlberg**
This should fix the randomly failing network table tests

### CODE REVIEW

File: core/src/test/java/edu/wpi/grip/core/operations/network/networktables/MockNetworkTable.java [hunk 1]
```diff
@@ -0,0 +1,506 @@
...
+    }
+    for (String p : key.split("/")) {
+      t = (MockNetworkTable) t.getSubTable(p);
+    }
+    return t;
+  }
+
+  @Override
+  public Set<String> getKeys(int types) {
+    return null;
```
Comment (SamCarlberg) [inline core/src/test/java/edu/wpi/grip/core/operations/network/networktables/MockNetworkTable.java]: this should throw `UnsupportedOperationException`

File: core/src/test/java/edu/wpi/grip/core/operations/network/networktables/MockNetworkTable.java [hunk 2]
```diff
@@ -0,0 +1,506 @@
...
+      byte[].class,
+      String.class,
+      String[].class,
+      Boolean.class,
+      Boolean[].class,
+      boolean[].class,
+      Double.class,
+      Double[].class,
+      double[].class
+  );
```
Comment (JLLeitschuh) [inline core/src/test/java/edu/wpi/grip/core/operations/network/networktables/MockNetworkTable.java]: You could use guava's immutable list here.

File: core/src/test/java/edu/wpi/grip/core/operations/network/networktables/MockNetworkTable.java [hunk 3]
```diff
@@ -0,0 +1,506 @@
...
+  }
+
+  @Override
+  public double getDouble(String key, double defaultValue) {
+    return getNumber(key, defaultValue);
+  }
+
+  @Override
+  public String toString() {
+    return "MockNetworkTable: " + name;
```
Comment (JLLeitschuh) [inline core/src/test/java/edu/wpi/grip/core/operations/network/networktables/MockNetworkTable.java]: This whole class really should be provided by network tables as a testing library. @PeterJohnson What do you think about that?

@SamCarlberg Keep it here though for this PR.

File: core/src/test/java/edu/wpi/grip/core/operations/network/networktables/MockNetworkTable.java [hunk 4]
```diff
@@ -0,0 +1,506 @@
...
+      String[].class,
+      Boolean.class,
+      Boolean[].class,
+      boolean[].class,
+      Double.class,
+      Double[].class,
+      double[].class
+  );
+
+  private static final MockNetworkTable ROOT = new MockNetworkTable("/");
```
Comment (JLLeitschuh) [inline core/src/test/java/edu/wpi/grip/core/operations/network/networktables/MockNetworkTable.java]: So, for each test we should have a fresh root?
I don't want tests to have static state that can pollute each other.

File: core/src/test/java/edu/wpi/grip/core/operations/network/networktables/MockNetworkTable.java (hunk 2 above)
Comment (SamCarlberg) [inline core/src/test/java/edu/wpi/grip/core/operations/network/networktables/MockNetworkTable.java]: They're both immutable; I just prefer to use official Java APIs

File: core/src/test/java/edu/wpi/grip/core/operations/network/networktables/MockNetworkTable.java (hunk 3 above)
Comment (SamCarlberg) [inline core/src/test/java/edu/wpi/grip/core/operations/network/networktables/MockNetworkTable.java]: Absolutely agree

File: core/src/test/java/edu/wpi/grip/core/operations/network/networktables/MockNetworkTable.java (hunk 4 above)
Comment (SamCarlberg) [inline core/src/test/java/edu/wpi/grip/core/operations/network/networktables/MockNetworkTable.java]: Remove in d967e78

**Comment by SamCarlberg** [comment]
@JLLeitschuh This is ready to be merged

**Comment by JLLeitschuh** [comment]
LGTM!

[Approved with PullApprove](https://pullapprove.com/WPIRoboticsProjects/GRIP/pull-request/675/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=WPIRoboticsProjects/GRIP)

**Pull request merged**